bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs

```diff
@@ -32,9 +33,20 @@ public static CryptClient Create(CryptOptions options)
 
             Status status = new Status();
 
+            // The below code can be avoided on Windows. So, we don't call it on this system 
+            // to avoid the restriction on target frameworks that should be equal to or higher than .netstandard2.1
+            if (OperationSystemHelper.CurrentOperationSystem != OperationSystemPlatform.Windows)
+            {
+                var cryptoCallback = new Library.Delegates.CryptoCallback(SigningRSAESPKCSCallback.RsaSign);
```

Comment:
To check my understanding, does this mean that C# relies on libmongocrypt's native crypto for all other crypto hooks? I don't see calls to `mongocrypt_setopt_crypto_hooks`.

Reply:
@markbenvenuto , can you please take a look at this question. I'm not familiar with why we don't use this `mongocrypt_setopt_crypto_hooks` hooks

Reply:
I don't see much point in only implement one crypt hook.

Reply:
1. The .NET driver is Windows only so it relies on the libmongocrypt support for Windows. It does not need the crypto hooks.
2. @jyemin My understanding is that the C# driver is looking to support non-Windows platforms which is why Dima did this work in preparation.

Reply:
it looks like we need to add support for `mongocrypt_setopt_crypto_hooks` as well. But I would suggest to do it in a new ticket. @jyemin what do you think?

Reply:
1. The .NET driver is not Windows only.  Many applications in fact run on Linux.  Even on Linux CSFLE should work if you configure things properly 
2. That said, we are looking to make CSFLE work more seamlessly on Linux, which will likely require all crypto hooks to be implemented.  But it doesn't help users just to implement just one, so might as well do them all at once.

Reply:
as It was discussed offline, I removed `mongocrypt_setopt_crypto_hook_sign_rsaes_pkcs1_v1_5`. This (and other hooks) should be added to the scope of linux support ticket